src/hooks/useUser.ts

```diff
@@ -14,14 +14,19 @@ type UseUserResult = {
   userId?: string;
 };
 
+const emptyUserProfile = {};
+
 export const useUser = (): UseUserResult => {
   const auth = useSelector(authSelector);
   const profile = useSelector(profileSelector);
 
   return {
     user: !auth.isEmpty ? auth : undefined,
     profile: !profile.isEmpty ? profile : undefined,
-    userWithId: auth && profile ? { ...profile, id: auth.uid } : undefined,
+    userWithId:
+      auth && profile
+        ? Object.assign(emptyUserProfile, profile, { id: auth.uid })
+        : undefined,
     userId: auth && profile ? auth.uid : undefined,
   };
```

Comment:
As written here, this is essentially mutating global state, and will have concurrency issues where literally every usage of the `useUser` hook is going to try and update the same object.

The proper way to fix this here is to `useMemo` the `userWithId` result, which seems like a worthwhile improvement to make in this PR, but it's not worth us using the verbose `Object.assign` syntax for in place of the current object spread syntax.

This is also unlikely to have a major impact on the performance issues across the platform, as most of those aren't coming via the current user, but by users read in through other places, which this will have no impact on.

Further Reading on object spread syntax:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals